checker/src/org/checkerframework/checker/nullness/NullnessChecker.java

```diff
@@ -1,6 +1,9 @@
 package org.checkerframework.checker.nullness;
 
+import org.checkerframework.framework.qual.StubFiles;
+
 /** A concrete instantiation of {@link AbstractNullnessChecker} using freedom-before-commitment. */
+@StubFiles("javadoc.astub")
```

Comment:
Parsing stub files is slow. Adding this new stub file here adds this overhead to everyone that wants to use the Nullness Checker, regardless of whether they use javadoc or not.
Is this really a good idea? Why not provide a script that produces an annotated `tools.jar`, similar to the process we use to produce an annotated `jdk8.jar`?
Then people that use javadoc can add that library, like they need to do for the other nullness-annotated libraries in `checker/lib`.

Reply:
The javadoc.astub file is only 100 lines long, so I hope the overhead isn't too high.
In the future moving the annotations into a proper library is a good idea, for these and all other stub files.  I don't think we should delay this pull request for that change, though.

Reply:
I would strongly prefer making it optional to include that file.
Nobody complained about the missing annotations. The only useful annotations are `@Pure`. I don't see anything nullness related.
I think this can wait until the annotations are cleaned up. If this important before then, we can include the file, but not include it for every run.

Reply:
> Nobody complained about the missing annotations.
Maybe not to you. :-)

Reply:
Ok, if you have a concrete user, they can use `-Astubs` to use them.

Reply:
Let's start this comment thread over.  The speed issue that sparked this thread is not important, but there is an important issue that motivates removing this `@StubFiles("javadoc.astub")` annotation.

If a user uses `-AstubDebug` but does not have `tools.jar` on the classpath, then there will be a flood of warnings about classes that are not found.  This is annoying and will hinder people in debugging their stub files.

Since a user has to specify `tools.jar` explicitly when compiling, it seems reasonable to require them to add an explicit stub file on the command line.

I had not thought about the problem of warnings about classes not found in stub files.  Sorry about suggesting use of `@StubFiles` here.